lms/models/organization.py

```diff
@@ -0,0 +1,75 @@
+from base64 import urlsafe_b64encode
+from uuid import uuid4
+
+import sqlalchemy as sa
+from sqlalchemy.dialects.postgresql import JSONB
+
+from lms.db import BASE
+from lms.models._mixins import CreatedUpdatedMixin
+from lms.models.application_settings import ApplicationSettings
+from lms.models.region import Region
+
+
+class Organization(CreatedUpdatedMixin, BASE):
+    """Model for Organizations comprised of application instances."""
+
+    __tablename__ = "organization"
+
+    id = sa.Column(sa.Integer(), autoincrement=True, primary_key=True)
+
+    _vanity_id = sa.Column(
+        sa.UnicodeText(),
+        nullable=False,
+        unique=True,
+        # We don't use a standard UUID-4 format here as they are common in Tool
+        # Consumer Instance GUIDs, and might be confused for them. These also
+        # happen to be shorter and guaranteed URL safe.
+        default=lambda: urlsafe_b64encode(uuid4().bytes).decode("ascii").rstrip("="),
+    )
+    """
+    A human readable URL safe vanity id.
+
+    The vanity id can be customised by the user and therefore can end-up not
+    being _globally_ unique, only locally unique to the LMS instance. For this
+    reason `public_id` should be used instead.
+    """
+
+    def public_id(self, region: Region) -> str:
+        """
+        Get the globally unique id which also indicates the region.
+
+        This returns vales with prefixes like 'us:' or 'ca:'.
+
+        This is useful if you only have the id, but don't know which region you
+        should be looking for it in and is the only id suitable for sharing
+        outside a single region LMS context.
+        """
+        return region.code + ":" + self._vanity_id
```

Comment:
This all looks like a bit of a palava, but I don't want to have any possibility of mixing the region code and the body of the id. This isn't something you can change as a user, so we don't want to get into trying to detect and split and join the values all the time.

I think this makes sense as this way you can have a normal form, and then write the value to the DB without much fuss. If it's unique and fits any other requirements you have (do we want to limit to certain chars?) then it's all golden.

Also, not sure we ever would, but this allows us to change the region code without doing much.